### DISCUSSION TITLE
Re-add `[unreleased]` suffix in changelog

### DISCUSSION THREAD

**Discussion by elenaf9** [libp2p/rust-libp2p#5781] [General]
We used to have an `[unreleased]` suffix in the changelog files if a version isn't released yet. It made it clear what the latest unreleased version is and easy to decide if a version bump is needed or not, without looking at crates.io.

We stopped adding this suffix, to avoid the additional work of removing it again upon release: https://github.com/libp2p/rust-libp2p/pull/4894.

However, I noticed that since then we've had quite a lot of unnecessary version bumps because it's not clear anymore if the version of a crate has been bumped since the last release, resulting in fix-up PRs like #5762 #5776 #5777 #5778 #5779 #5780.

The CI checks if a version has been bumped since the last released and an CHANGELOG entry added, but it doesn't check that the version has _only been bumped once_.

To avoid this in the future, I think I would prefer if we go back to having the `[unreleased]` suffix, even if that then requires some extra work when releasing. Wdyt @jxs @dariusc93? Alternative suggestions are also welcome.

**Comment by dariusc93** [comment]
I dont mind going back to having the suffix. It could then be removed when preparing to release/publish the corresponding crate like previously done. I am also not against looking into having CI catching unpublished crates when updating changelog (though not sure if this would be desirable). I think the former would be better though since it would give noticed about unpublished versions.

**Reply by drHuangMHT** [reply]
I prefer the suffix because that gives the information directly to whoever is reading the changelog without looking through CI results. But catching unpublished crates in CI is also a nice-to-have line of defense.

**Reply by dariusc93** [reply]
Might be best to go back to using it then :)

**Reply by elenaf9** [reply]
@jxs and I had an out-of-band discussion about this. Summary is that there is a preference to not have to touch the repo when releasing, which would be needed if we had the `unreleased` suffix.
I am working right now on a CI check to catch an invalid version bump, see #5804.

**Reply by drHuangMHT** [reply]
>  Summary is that there is a preference to not have to touch the repo when releasing, we would be needed if we had the `unreleased` suffix.

I thought the release process is somewhat automated.
